Static link failure with LLVM, RISCV cross-compilation

### Describe the bug
Linker bugs out (claiming bad flag `-lc`) when attempting to statically cross-compile for riscv64 using LLVM15/16 (not sure about other LLVM versions; haven't tried).

### Steps To Reproduce

Set up the following flake.nix

```nix
{
inputs = {
    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
    flake-utils.url = "github:numtide/flake-utils";
  };

  outputs = { self, nixpkgs, flake-utils }:
    flake-utils.lib.eachDefaultSystem
      (system:
      let pkgs = import nixpkgs { inherit system; };
      in rec {
        riscv-matmul-llvm = (pkgs.makeStatic pkgs.pkgsCross.riscv64.llvmPackages_16.stdenv).mkDerivation {
          NIX_DEBUG = 6;
          pname = "riscv-matmul-llvm";
          version = "0.0.1";
          src = ./out;
          buildPhase = ''
            $CC -march=rv64gcv -o riscv-matmul-llvm main.c
          '';
          installPhase = ''
            mkdir -p $out/bin
            cp riscv-matmul-llvm $out/bin
          '';
        };
        defaultPackage = riscv-matmul-llvm;
      });
}
```

and the following file in out/main.c:

```c
#include <stdio.h>
#include <time.h>

int main() {
  clock_t start, end;
  int msec;
  static int M = 29;
  float A[M];
  
  start = clock();
  for (int q = 0; q < 1000; q++)
    for (int i = 0; i < M; i++)
        A[i] = 3.2;
  end = clock();

  msec = (end - start) * 1000;
  printf("Time taken: %d milliseconds\n", msec);

  return (0);
}
```

and run `nix build`.

The compilation bugs out with the following error:

```
/nix/store/rvf7rdpfcszl4wsl7lsaqn7f6arsb5wp-riscv64-unknown-linux-gnu-binutils-2.40/bin/riscv64-unknown-linux-gnu-ld: cannot find -lc: No such file or directory
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
```

The full error log (along with flake lockfile) can be found [here](https://gist.github.com/gdinh/036aad090628aeeaf2ec55f83d24f3db).

The same error also occurs with LLVM 15.

This error does not occur with dynamic linking (deleting `pkgs.makeStatic`); everything compiles fine in that case.

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.0-153-generic, Ubuntu, 20.04.6 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.16.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/home/eecs/dinh/.nix-defexpr/channels/nixpkgs`
```

Pinging @amjoseph-nixpkgs at the suggeston of @Artturin.

--- a/pkgs/stdenv/adapters.nix
+++ b/pkgs/stdenv/adapters.nix
@@ -67,7 +67,7 @@ rec {
             "--disable-shared" # brrr...
           ];
       }));
-    } // lib.optionalAttrs (stdenv0.hostPlatform.libc == "libc") {
+    } // lib.optionalAttrs (stdenv0.hostPlatform.libc == "glibc") {
       extraBuildInputs = (old.extraBuildInputs or []) ++ [
         pkgs.glibc.static
       ];

--- a/pkgs/stdenv/adapters.nix
+++ b/pkgs/stdenv/adapters.nix
@@ -121,9 +121,6 @@ rec {
     # Apple does not provide a static version of libSystem or crt0.o
     # So we can’t build static binaries without extensive hacks.
     ++ lib.optional (!stdenv.hostPlatform.isDarwin) makeStaticBinaries
-
-    # Glibc doesn’t come with static runtimes by default.
-    # ++ lib.optional (stdenv.hostPlatform.libc == "glibc") ((lib.flip overrideInStdenv) [ self.glibc.static ])
   );
 
 

